Content.Server/Construction/ConstructionSystem.cs

```diff
@@ -94,6 +94,12 @@ private IEnumerable<IEntity> EnumerateNearby(IEntity user)
 
         private async Task<IEntity?> Construct(IEntity user, string materialContainer, ConstructionGraphPrototype graph, ConstructionGraphEdge edge, ConstructionGraphNode targetNode)
         {
+            if (user.IsInContainer())
```

Comment:
You should put this in `HandleStartStructureConstruction` instead. Crafting items inside a container should be fine.

Reply:
I've moved it to HandleStartStructureConstruction after its user null check to not make more of those than necessary

Suppose it could go before the TryIndecies but that enters into the domain of taste afaik